File: src/translator/zh2en/provider/attribute.ts
import { DATA as POE_DATA } from "../../../data/poe/index.js";
import type { Attribute } from "../../../data/poe/types.js";

export class AttributeProvider {
    private zhIdx = new Map<string, Attribute>();

    constructor() {
        for (const p of POE_DATA.attributes) {
            const zh = p.zh;
            this.zhIdx.set(zh, p);
        }
    }

    provideByZh(zh: string): Attribute | undefined {
        return this.zhIdx.get(zh);
    }
}

File: go/translator/zh2en/provider/attribute.go
package provider

import "github.com/cn-poe-community/cn-poe-utils/go/data/poe"

type AttributeProvider struct {
	zhIdx map[string]*poe.Attribute
}

func NewAttributeProvider() *AttributeProvider {
	zhIdx := make(map[string]*poe.Attribute)
	for i := range poe.DATA.Attributes {
		attr := &poe.DATA.Attributes[i]
		zhIdx[attr.Zh] = attr
	}
	return &AttributeProvider{
		zhIdx: zhIdx,
	}
}

func (p *AttributeProvider) ProvideByZh(zh string) *poe.Attribute {
	if attr, ok := p.zhIdx[zh]; ok {
		return attr
	}

	return nil
}
